test/e2e/admin/admin-checkout.test.ts

```diff
@@ -0,0 +1,317 @@
+import * as firebase from 'firebase';
+import { slow, suite, test } from 'mocha-typescript';
+import squel from 'squel';
+import { CheckoutItems } from '../../../src/models/checkout-items/checkout-items';
+import { CheckoutObject } from '../../../src/models/checkout-object/checkout-object';
+import { IntegrationTest } from '../integration-test';
+import { TestData } from '../test-data';
+
+let listener: firebase.Unsubscribe;
+let firebaseUser: firebase.User;
+
+function login(email: string, password: string): Promise<firebase.User> {
+  if (firebaseUser) {
+    return new Promise(resolve => resolve(firebaseUser));
+  }
+  return new Promise((resolve, reject) => {
+    firebase.auth()
+      .signInWithEmailAndPassword(email, password)
+      .catch(err => reject(err));
+    listener = firebase.auth()
+      .onAuthStateChanged((user) => {
+        if (user) {
+          firebaseUser = user;
+          resolve(user);
+        }
+      });
+  });
+}
+
+function loginAdmin() {
```

Comment:
Is this the only file that needs firebase auth? Can it be moved to a central helper and imported in other files?

Reply:
Almost all the routes require an idToken so all the integration test files need firebase auth, but since they all need it, then yeah it can be moved to a central helper and maybe inherited.